src/ts/app/framework/src/improved-loop-for-testing.ts

```diff
@@ -0,0 +1,48 @@
+import {ImprovedLoop} from "app/framework"
+import {Action} from "redux"
+import {ImprovedLoopAction, ImprovedLoopPromiseFunc} from "./improved-loop"
+
+export class ImprovedLoopForTesting<DispatchedA extends Action> implements ImprovedLoop<any, any, DispatchedA> {
```

Comment:
this should probably be a -test.ts, and in a separate project called test-helpers (or something). We want to make sure test code doesn't get accidentally included into production code, and tsconfig is able to manage that.